cupy/core/_routines_indexing.pyx

```diff
@@ -49,6 +49,16 @@ cdef _ndarray_setitem(ndarray self, slices, value):
 
 
 cdef tuple _ndarray_nonzero(ndarray self):
+    cdef int ndim
+    dst = _ndarray_argwhere(self)
```

Comment:
Since we know the type, this should generate more efficient C code in the end.

```suggestion
    cdef ndarray dst = _ndarray_argwhere(self)
```

Reply:
done